apps/test/unit/StudioAppTest.js

```diff
@@ -1,98 +1,131 @@
-          blocks: [],
-        },
-        skin: {},
+    beforeEach(() => {
+      stubRedux();
+      registerReducers(commonReducers);
+    });
+    afterEach(restoreRedux);
+
+    describe("the init() method", () => {
+      let files;
+      beforeEach(() => {
+        files = [];
+        sinon.stub(studioApp(), 'configureDom');
+        sinon.stub(assetsApi, 'getFiles').callsFake(cb => cb({files}));
+        sinon.spy(listStore, 'reset');
       });
 
-      expect(assetsApi.getFiles).to.have.been.calledOnce;
-      expect(listStore.reset).to.have.been.calledWith(files);
-    });
+      afterEach(() => {
+        assetsApi.getFiles.restore();
+        listStore.reset.restore();
+      });
 
-    it("will emit an afterInit event", () => {
-      const listener = sinon.spy();
-      studioApp().on('afterInit', listener);
-      studioApp().init({
-        usesAssets: true,
-        enableShowCode: true,
-        containerId: 'foo',
-        level: {
-          editCode: true,
-          codeFunctions: {},
-        },
-        dropletConfig: {
-          blocks: [],
-        },
-        skin: {},
+      it('will pre-populate assets for levels that use assets', () => {
+        studioApp().init({
+          usesAssets: true,
+          enableShowCode: true,
+          containerId: 'foo',
+          level: {
+            editCode: true,
+            codeFunctions: {},
+          },
+          dropletConfig: {
+            blocks: [],
+          },
+          skin: {},
+        });
+
+        expect(assetsApi.getFiles).to.have.been.calledOnce;
+        expect(listStore.reset).to.have.been.calledWith(files);
       });
 
-      expect(listener).to.have.been.calledOnce;
+      it("will emit an afterInit event", () => {
+        const listener = sinon.spy();
+        studioApp().on('afterInit', listener);
+        studioApp().init({
+          usesAssets: true,
+          enableShowCode: true,
+          containerId: 'foo',
+          level: {
+            editCode: true,
+            codeFunctions: {},
+          },
+          dropletConfig: {
+            blocks: [],
+          },
+          skin: {},
+        });
+
+        expect(listener).to.have.been.calledOnce;
+      });
+    });
+  });
+
+  describe('StudioApp.makeFooterMenuItems', () => {
+    beforeEach(() => {
+      sinon.stub(project, 'getUrl');
+    });
+
+    afterEach(() => {
+      project.getUrl.restore();
+    });
+
+    const i18n = {
+      t: key => key
+    };
+
+    it("How It Works URL for game lab embed page", () => {
+      project.getUrl.returns('https://studio.code.org/projects/gamelab/C_2x38fH_jElONWxTLrCHw/embed');
+      const footItems = makeFooterMenuItems(i18n, project);
+      const howItWorksItem = footItems.find(item =>
+          item.text === 'footer.how_it_works');
+      expect(howItWorksItem.link).to.equal('https://studio.code.org/projects/gamelab/C_2x38fH_jElONWxTLrCHw/edit');
+    });
+
+    it("How It Works URL for game lab share page", () => {
```

Comment:
When you run the unit test suite, it will spit out test results using the strings given to `describe()` and `it()`. I like to try and make these strings translate into human readable sentences, such that you could concatenate all the `describe` strings and the current `it` string into a single sentence. Here's what it would look like as is:

```
StudioApp.makeFooterMenuItems How It Works URL for game lab embed page
StudioApp.makeFooterMenuItems How It Works URL for game lab share page
```

How about this instead:

```
The StudioApp.makeFooterMenuItems function returns a "How It Works" link pointing to the project edit page when on the embed page
The StudioApp.makeFooterMenuItems function returns a "How It Works" link pointing to the project edit page when on the share page
```

Then if someone makes the test fail, it's easy to understand what behavior is being tested without having to read the test itself.